src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py

```diff
@@ -228,15 +229,129 @@ def acr_repository_show_manifests(cmd,
         result_index='manifests')
 
 
+def acr_repository_untag(cmd,
+                         registry_name,
+                         image,
+                         resource_group_name=None,
+                         username=None,
+                         password=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, UNTAG_NOT_SUPPORTED)
+
+    repository, tag, _ = _parse_image_name(image)
+
+    login_server, username, password = get_access_credentials(
+        cli_ctx=cmd.cli_ctx,
+        registry_name=registry_name,
+        resource_group_name=resource_group_name,
+        username=username,
+        password=password,
+        repository=repository,
+        permission='*')
+
+    return _delete_data_from_registry(
+        login_server=login_server,
+        path='/v2/_acr/{}/tags/{}'.format(repository, tag),
+        username=username,
+        password=password)
+
+
 def acr_repository_delete(cmd,
                           registry_name,
-                          repository,
+                          repository=None,
+                          image=None,
                           tag=None,
                           manifest=None,
                           resource_group_name=None,
                           username=None,
                           password=None,
                           yes=False):
+    if (not repository and not image) or (repository and image):
+        raise CLIError("Please specify either an image name with --image or a repository name with --repository.")
+
+    # Check if this is a legacy command. --manifest can be used as a flag so None is checked.
+    if repository and (tag or manifest is not None):
+        return _legacy_delete(cmd=cmd,
+                              registry_name=registry_name,
+                              repository=repository,
+                              tag=tag,
+                              manifest=manifest,
+                              resource_group_name=resource_group_name,
+                              username=username,
+                              password=password,
+                              yes=yes)
+
+    if tag:
+        logger.warning("The parameter --tag is deprecated. The specified --tag will be ignored.")
+    if manifest:
+        logger.warning("The parameter --manifest is deprecated. The specified --manifest will be ignored.")
```

Comment:
Okay, I like that you have the warning here, but it seems like this would be the breaking change you are trying to avoid?  If you ignore manifest or tag, then you may as well mark them ignore so they can't be specified.  If a script used these, to have the script not explicitly fail but just not work is still a breaking change (and I would argue a worse one). I would suggest, if you are trying to avoid a breaking change:
1) The command should still support these commands for X number of releases.
2) The deprecation message should point them to the right parameter or command so they can update their scripts prior to the functionality being completely removed.

Reply:
If the code reaches here. The user must have specified the new parameter `--image` (which contains `tag` or `manifest`) with `--tag` or `--manifest` parameters, so we ignore them and just use `--image`. We could error out here too if that's preferred. These two warnings don't affect deprecated commands that are handled right before here:

```
    if repository and (tag or manifest is not None):
      ...
```

The goal of this PR is to have backward compatibility while adding the new `--image` parameter.

Reply:
Added more comments to hopefully clarify the logic here.

Reply:
Thanks for the clarification. To avoid someone creating a bad script, my recommendation would be to throw an error in this situation.